src/js/controller/captions.js

```diff
@@ -57,17 +57,20 @@ const Captions = function(_model) {
     }, this);
 
     function _setSubtitlesTracks(tracks) {
-        if (!tracks.length) {
+        if (!Array.isArray(tracks)) {
             return;
         }
+        if (!tracks.length) {
+            _tracks = [];
```

Comment:
Shouldn't `_tracksById` be reset here too?

These properties should be set together:

```
        _tracks = [];
        _tracksById = {};
        _unknownCount = 0;
```

Reply:
Updated!